modules/OpenXRJoypad_module/src/OpenXRJoypadModule.cpp

```diff
@@ -0,0 +1,1384 @@
+
+    /**
+     * Check if the velocity control is used
+     * @return true if the velocity control is used
+     */
+    bool isVelocityControlUsed();
+
+    /**
+     * Get feedback from the robot
+     * @return true / false in case of success / failure
+     */
+    bool getFeedback();
+
+    /**
+     * Get the joint limits
+     * @param limits matrix containing the joint limits in radian
+     * @return true / false in case of success / failure
+     */
+    bool getLimits(yarp::sig::Matrix& limits);
+
+    /**
+     * Get the joint encoders value
+     * @return the joint encoder value
+     */
+    const yarp::sig::Vector& jointEncoders() const;
+
+    /**
+     * Get the number of degree of freedom
+     * @return the number of actuated DoF
+     */
+    int getDoFs();
+
+    /**
+     * Close the helper
+     */
+    void close();
+};
+
+bool RobotControlHelper::configure(const yarp::os::Searchable& config, const std::string& name)
+{
+
+    // robot name: used to connect to the robot
+    std::string robot;
+    robot = config.check("robot", yarp::os::Value("icubSim")).asString();
+
+    // get all controlled icub parts from the resource finder
+    std::vector<std::string> iCubParts;
+    yarp::os::Value* iCubPartsYarp;
+    if (!config.check("remote_control_boards", iCubPartsYarp))
+    {
+        yError() << "[RobotControlHelper::configure] Unable to find remote_control_boards into "
+                    "config file.";
+        return false;
+    }
+    if (!YarpHelper::yarpListToStringVector(iCubPartsYarp, iCubParts))
+    {
+        yError() << "[RobotControlHelper::configure] Unable to convert yarp list into a vector of "
+                    "strings.";
+        return false;
+    }
+
+    // open the remotecontrolboardremepper YARP device
+    yarp::os::Property options;
+    yarp::os::Value* axesListYarp;
+    if (!config.check("joints_list", axesListYarp))
+    {
+        yError() << "[RobotControlHelper::configure] Unable to find joints_list into config file.";
+        return false;
+    }
+
+    if (!YarpHelper::yarpListToStringVector(axesListYarp, m_axesList))
+    {
+        yError() << "[RobotControlHelper::configure] Unable to convert yarp list into a "
+                    "vector of strings.";
+        return false;
+    }
+
+    options.put("device", "remotecontrolboardremapper");
+    YarpHelper::addVectorOfStringToProperty(options, "axesNames", m_axesList);
+
+    // prepare the remotecontrolboards
+    yarp::os::Bottle remoteControlBoards;
+    remoteControlBoards.clear();
+    yarp::os::Bottle& remoteControlBoardsList = remoteControlBoards.addList();
+    for (auto& iCubPart : iCubParts)
+        remoteControlBoardsList.addString("/" + robot + "/" + iCubPart);
+
+    options.put("remoteControlBoards", remoteControlBoards.get(0));
+    options.put("localPortPrefix", "/" + name + "/remoteControlBoard");
+    yarp::os::Property& remoteControlBoardsOpts = options.addGroup("REMOTE_CONTROLBOARD_OPTIONS");
```

Comment:
Not now but we may think of using blf in the future

Reply:
This is indeed a good point. TBH I copied some old existing code and I did not invest too much time on it. I have opened https://github.com/robotology/walking-teleoperation/issues/134 as a reminder